### PR TITLE
Common Quantities first draft

### PR DESCRIPTION
**Pull request by AlexBrad1eyCT**
### Developments

QSets for common schema elements and some road elements that are needed. Drafted by myself and discussed at meeting on 30th March 2021.

### CODE REVIEW

**Comment by AlexBrad1eyCT** [comment]
Any movement on getting these reviewed???

**Comment by SergejMuhic** [comment]
Sorry for leaving this unattended for so long. Had to resolve the issues with Jon first. Will do address it this week.

File: IFC4x3/Properties/n/NominalDepth_0HZnE0qUaHuO00025QrE$V/DocProperty.xml [hunk 1]
```diff
@@ -7,3 +7,4 @@
 		</DocLocalization>
 	</Localization>
 </DocProperty>
+
```
Comment (SergejMuhic) [inline IFC4x3/Properties/n/NominalDepth_0HZnE0qUaHuO00025QrE$V/DocProperty.xml:10]: This is a weird empty line that snuck in. @jmirtsch do you think this is a bug?

File: IFC4x3/Quantities/c/CompactedVolume_1clL3F_GzEROzZL0CFLOfa/DocQuantity.xml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (SergejMuhic) [inline IFC4x3/Quantities/c/CompactedVolume_1clL3F_GzEROzZL0CFLOfa/DocQuantity.xml:1]: Quantity type is missing

File: IFC4x3/Quantities/l/LooseVolume_1f8iHGSB91FBK9qA7FuE68/DocQuantity.xml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocQuantity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="LooseVolume_1f8iHGSB91FBK9qA7FuE68" Name="LooseVolume" UniqueId="6922c450-70b2-413c-b509-d0a1cfe0e188" />
```
Comment (SergejMuhic) [inline IFC4x3/Quantities/l/LooseVolume_1f8iHGSB91FBK9qA7FuE68/DocQuantity.xml:2]: Quantity type missing

File: IFC4x3/Quantities/s/SignArea_32i2POqiz7sxewNKXzANau/DocQuantity.xml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocQuantity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="SignArea_32i2POqiz7sxewNKXzANau" Name="SignArea" UniqueId="c2b02658-d2cf-47db-ba3a-5d487d297938" />
```
Comment (SergejMuhic) [inline IFC4x3/Quantities/s/SignArea_32i2POqiz7sxewNKXzANau/DocQuantity.xml:2]: Quantity type missing

File: IFC4x3/Quantities/u/UndisturbedVolume_26rNXiavD40gy60NShbzTH/DocQuantity.xml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocQuantity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="UndisturbedVolume_26rNXiavD40gy60NShbzTH" Name="UndisturbedVolume" UniqueId="86d5786c-9393-4402-af06-01772b97d751" />
```
Comment (SergejMuhic) [inline IFC4x3/Quantities/u/UndisturbedVolume_26rNXiavD40gy60NShbzTH/DocQuantity.xml:2]: Quantity type missing

File: IFC4x3/Quantities/w/Weight_2ueoe8QyrBduWuJHsi21zC/Documentation.md [hunk 6]
```diff
@@ -0,0 +1 @@
+Total weight of object
```
Comment (SergejMuhic) [inline IFC4x3/Quantities/w/Weight_2ueoe8QyrBduWuJHsi21zC/Documentation.md:1]: Add punctuation.

File: IFC4x3/Quantities/w/Weight_2ueoe8QyrBduWuJHsi21zb/Documentation.md [hunk 7]
```diff
@@ -1 +1 @@
-Total weight of reinforcing.
+Total weight of object
```
Comment (SergejMuhic) [inline IFC4x3/Quantities/w/Weight_2ueoe8QyrBduWuJHsi21zb/Documentation.md:1]: Wondering how we have two Documentation files for weight.

File: IFC4x3/Quantities/t/Thickness_29hrivl4D9Xv3Ij$kS6FDr/Documentation.md [hunk 8]
```diff
@@ -0,0 +1 @@
+the geometric thickness of the object
```
Comment (SergejMuhic) [inline IFC4x3/Quantities/t/Thickness_29hrivl4D9Xv3Ij$kS6FDr/Documentation.md:1]: Capital + punctuation

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_SignBaseQuantities/DocQuantitySet.xml [hunk 9]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocQuantitySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Qto_SignBaseQuantities" UniqueId="fb485ee8-9c7e-46ca-9d0f-eac5dfd7aa41">
+	<Quantities>
+		<DocQuantity xsi:nil="true" href="Height_3jq0d8r0zBYvccpej6okcm" />
+		<DocQuantity xsi:nil="true" href="Width_2vkF8AsQL5fg4keqWbdE8G" />
+		<DocQuantity xsi:nil="true" href="Thickness_29hrivl4D9Xv3IjkS6FDr" />
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_SignBaseQuantities/DocQuantitySet.xml:6]: Thickness surprised me here a bit.

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_SignalBaseQuantities/DocQuantitySet.xml [hunk 10]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocQuantitySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Qto_SignalBaseQuantities" UniqueId="b02c9d87-835c-43bb-bedb-7b4a65013060">
+	<Quantities>
+		<DocQuantity xsi:nil="true" href="Weight_2ueoe8QyrBduWuJHsi21zb" />
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_SignalBaseQuantities/DocQuantitySet.xml:4]: Does it make sense to add height and width here?

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_PictorialSignQuantities/DocQuantitySet.xml [hunk 11]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocQuantitySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Qto_PictorialSignQuantities" UniqueId="c0785096-718a-459d-8ae8-b76089be6d06">
+	<Quantities>
+		<DocQuantity xsi:nil="true" href="Area_1CnNsFYBT6AemE3IaDxL7" />
+		<DocQuantity xsi:nil="true" href="SignArea_32i2POqiz7sxewNKXzANau"/>
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_PictorialSignQuantities/DocQuantitySet.xml:5]: Height and width of interest?

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_ImpactProtectionDeviceBaseQuantities/DocQuantitySet.xml [hunk 12]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocQuantitySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Qto_ImpactProtectionDeviceBaseQuantities" UniqueId="4edbaef5-6f65-4131-a5b8-1772029277a8">
+	<Quantities>
+		<DocQuantity xsi:nil="true" href="Weight_2ueoe8QyrBduWuJHsi21zb" />
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_ImpactProtectionDeviceBaseQuantities/DocQuantitySet.xml:4]: Interesting one. Seems like it needs something more but I cannot put my finger on it. @larswik would like to hear your opinion also.

**Review by SergejMuhic — CHANGES_REQUESTED** [review]
Some of the q types were missing. Might have missed something in the description though.

File: IFC4x3/Properties/n/NominalDepth_0HZnE0qUaHuO00025QrE$V/DocProperty.xml (hunk 1 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Properties/n/NominalDepth_0HZnE0qUaHuO00025QrE$V/DocProperty.xml:10]: Weird...

File: IFC4x3/Quantities/c/CompactedVolume_1clL3F_GzEROzZL0CFLOfa/DocQuantity.xml (hunk 2 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Quantities/c/CompactedVolume_1clL3F_GzEROzZL0CFLOfa/DocQuantity.xml:1]: fixed

File: IFC4x3/Quantities/s/SignArea_32i2POqiz7sxewNKXzANau/DocQuantity.xml (hunk 4 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Quantities/s/SignArea_32i2POqiz7sxewNKXzANau/DocQuantity.xml:2]: fixed

File: IFC4x3/Quantities/u/UndisturbedVolume_26rNXiavD40gy60NShbzTH/DocQuantity.xml (hunk 5 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Quantities/u/UndisturbedVolume_26rNXiavD40gy60NShbzTH/DocQuantity.xml:2]: fixed

File: IFC4x3/Quantities/w/Weight_2ueoe8QyrBduWuJHsi21zC/Documentation.md (hunk 6 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Quantities/w/Weight_2ueoe8QyrBduWuJHsi21zC/Documentation.md:1]: this is a phantom quantity andf is will be deleted

File: IFC4x3/Quantities/w/Weight_2ueoe8QyrBduWuJHsi21zb/Documentation.md (hunk 7 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Quantities/w/Weight_2ueoe8QyrBduWuJHsi21zb/Documentation.md:1]: the other on is a phantom property andf will be deleted

File: IFC4x3/Quantities/t/Thickness_29hrivl4D9Xv3Ij$kS6FDr/Documentation.md (hunk 8 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Quantities/t/Thickness_29hrivl4D9Xv3Ij$kS6FDr/Documentation.md:1]: fixed

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_SignBaseQuantities/DocQuantitySet.xml (hunk 9 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_SignBaseQuantities/DocQuantitySet.xml:6]: well the sign plate has a thickness and if its a smart sign they are quite chunky! do you prefer depth?

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_PictorialSignQuantities/DocQuantitySet.xml (hunk 11 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_PictorialSignQuantities/DocQuantitySet.xml:5]: that is in the entity qto  `Qto_SignBaseQuantities` these are additional qtos for pictoral signs as the entity includes markers which are just posts

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_ImpactProtectionDeviceBaseQuantities/DocQuantitySet.xml (hunk 12 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_ImpactProtectionDeviceBaseQuantities/DocQuantitySet.xml:4]: it is a percular one the geometrics possibilities are so varied that anything other than weight could have a vast diference in meaning depending on the particular type of device.

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_SignalBaseQuantities/DocQuantitySet.xml (hunk 10 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_SignalBaseQuantities/DocQuantitySet.xml:4]: difficult one its similar to my impact device comment the geometric meaning of the height and width is very diffenent depending on the particular modelled signal

File: IFC4x3/Quantities/l/LooseVolume_1f8iHGSB91FBK9qA7FuE68/DocQuantity.xml (hunk 3 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Quantities/l/LooseVolume_1f8iHGSB91FBK9qA7FuE68/DocQuantity.xml]: fixed

**Comment by AlexBrad1eyCT** [comment]
Fixed all the errors and found some missing documentation, just the philisophical conversations to resolve!

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_SignBaseQuantities/DocQuantitySet.xml (hunk 9 above)
Comment (SergejMuhic) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_SignBaseQuantities/DocQuantitySet.xml:6]: My surprise was more in the line of when would it be a required quantity. Is it something that is required in the chunky signs? For operation, I would not expect this, Maybe in operation?

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_PictorialSignQuantities/DocQuantitySet.xml (hunk 11 above)
Comment (SergejMuhic) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_PictorialSignQuantities/DocQuantitySet.xml]: I see.

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_ImpactProtectionDeviceBaseQuantities/DocQuantitySet.xml (hunk 12 above)
Comment (SergejMuhic) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_ImpactProtectionDeviceBaseQuantities/DocQuantitySet.xml:4]: How about just having something like nominal dimensions? i.s. bounding box.

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_SignBaseQuantities/DocQuantitySet.xml (hunk 9 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_SignBaseQuantities/DocQuantitySet.xml:6]: it a BOQ value construction need to know what material to get to make the sign out of eg thickness of steel plate to buy.

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_ImpactProtectionDeviceBaseQuantities/DocQuantitySet.xml (hunk 12 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_ImpactProtectionDeviceBaseQuantities/DocQuantitySet.xml:4]: is that valid within the sense of a quantity? to specify  the bounding box size?

Comment (SergejMuhic) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/QuantitySets/Qto_ImpactProtectionDeviceBaseQuantities/DocQuantitySet.xml:4]: Not really, that is why I am tippy-toeing around it. :)

File: IFC4x3/Quantities/v/Volume_0v4MMyo9v59fK0AkoEQN9i/Documentation.md [hunk 13]
```diff
@@ -1 +1 @@
-Volume of the opening. It is the addition volume of the project to the element (e.g. wall or slab).
+Volume of the opening. It is the addition volume of the project to the element (e.g. wall or slab).
```
[the remark this answers is not in the document]
Comment (AlexBrad1eyCT) [reply IFC4x3/Quantities/v/Volume_0v4MMyo9v59fK0AkoEQN9i/Documentation.md:1]: This is an original description that for some reason has changed. (but not changed) i have updated the language to be understandable.

**Comment by jmirtsch** [comment]
I conducted a quick review and looks good to me.  Happy for this to be merged.

**Pull request merged**